x/auctions/keeper/hooks_test.go

```diff
@@ -0,0 +1,143 @@
+// Copyright Tharsis Labs Ltd.(Evmos)
+// SPDX-License-Identifier:ENCL-1.0(https://github.com/evmos/evmos/blob/main/LICENSE)
+
+package keeper_test
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	govtypes "github.com/cosmos/cosmos-sdk/x/gov/types"
+	"github.com/evmos/evmos/v19/utils"
+	epochstypes "github.com/evmos/evmos/v19/x/epochs/types"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	testkeyring "github.com/evmos/evmos/v19/testutil/integration/evmos/keyring"
+	testnetwork "github.com/evmos/evmos/v19/testutil/integration/evmos/network"
+	testutiltx "github.com/evmos/evmos/v19/testutil/tx"
+	"github.com/evmos/evmos/v19/x/auctions/types"
+)
+
+func TestHookAfterEpochEnd(t *testing.T) {
+	var network *testnetwork.UnitTestNetwork
+
+	validSenderAddr, _ := testutiltx.NewAccAddressAndKey()
+	// Token used in the bid.
+	bidAmount := sdk.NewCoin(utils.BaseDenom, sdk.NewInt(1))
+	// Tokens used to simulate auctioned tokens.
+	additionalCoins := []sdk.Coin{
+		sdk.NewInt64Coin("atoken", 1),
+		sdk.NewInt64Coin("btoken", 1),
+	}
+	// Coin used to simulate accrued tokens to auction.
+	feeCoin := sdk.NewInt64Coin("fee", 1)
+
+	// Value required in the hook but not used.
+	unusedEpochNumber := int64(0)
+
+	testCases := []struct {
+		name            string
+		malleate        func()
+		epochIdentifier string
+		postCheck       func()
+	}{
+		{
+			name: "pass with prize distributed",
```

Comment:
The two tests have the same malleate because the way in which we are using hook does not allows to check for errors. If we want to allow also errors as output, here we will add more tests.